src/main/kotlin/org/jetbrains/kotlinx/jupyter/repl/impl/InternalEvaluatorImpl.kt

```diff
@@ -124,4 +158,73 @@ internal class InternalEvaluatorImpl(
             isExecuting = false
         }
     }
+
+    private fun updateVariablesState(cellId: Int) {
+        // remove known modifying usages in this cell
+        usageMap[cellId]?.removeIf {
+            containingMap[it] != cellId
+        }
+
+        variablesMap.forEach {
+            val state = it.value
+            val property = state.propertyKlass
+            val newValue = property.get(state.scriptInstance) ?: return@forEach
+
+            if (state.stringValue != newValue.toString()) {
+                usageMap[cellId]?.add(it.key)
+                state.stringValue = newValue.toString()
+            }
+        }
+    }
+
+    private fun getVisibleVariables(target: ResultValue, cellId: Int): Map<String, VariableState> {
+        val klass = target.scriptClass ?: return emptyMap()
```

Comment:
Please use `kClass` or `class` word in naming

Reply:
Seems like a naming convention, doesn't it?

Reply:
If you have any references, please share. Common replacement of `class` is `clazz`, but if it's a part of the identifier, you don't have to replace it.

Reply:
No, I do not have references, just tried to name class) But stick with a kClass